resolver/resolver.go

```diff
@@ -17,6 +17,10 @@ type Resolver interface {
 	// an error. If the provided context is canceled, the resolution is also
 	// canceled.
 	Resolve(context.Context, string) (string, error)
+	// Upgrade resolves the given reference to the most recent release version,
+	// returning the resolved reference or an error. If the provided context is
+	// canceled, the resolution is also canceled.
+	Upgrade(context.Context, string) (string, error)
```

Comment:
I wonder if we should make a new interface instead? Then we can do the type assertion on whether the resolver can handle upgrades instead of stubbing out `Upgrade` in unsupported resolvers.